src/engine/validations/validate-locales.ts

```diff
@@ -0,0 +1,25 @@
+import { ApplyChangesetContext } from '../apply-changeset/types'
+import { AuthorizationErrorForApply } from '../errors'
+
+export async function validateLocales(context: ApplyChangesetContext) {
+  const sourceEnvironmentId = context.changeset.sys.source.sys.id
+  let sourceLocales, targetLocales
+
+  try {
+    ;[sourceLocales, targetLocales] = await Promise.all([
+      context.client.cma.locales.getMany({ environment: sourceEnvironmentId }),
+      context.client.cma.locales.getMany({ environment: context.environmentId }),
+    ])
+  } catch (err) {
+    throw new AuthorizationErrorForApply(context)
```

Comment:
This assumes that the only error possible is a 401, is that what we want?

Reply:
adjusted 👍